test/factory.ts

```diff
@@ -0,0 +1,54 @@
+
+import { ethers } from "hardhat";
+import { Contract, ContractFactory } from "@ethersproject/contracts";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+
+const REFUNDER_VERSION = 1;
+
+describe('Factory', () => {
+    let masterRefunder: Contract;
+    let factory: Contract;
+
+    let owner: SignerWithAddress;
+	let notOwner: SignerWithAddress;
+
+    let Refunder: ContractFactory;
+
+    beforeEach(async () => {
+        [owner, notOwner] = await ethers.getSigners(); 
+
+        Refunder = await ethers.getContractFactory("Refunder");
+		masterRefunder = await Refunder.deploy();
+		await masterRefunder.deployed();
+        await masterRefunder.init(owner.address);
+
+        let Registry = await ethers.getContractFactory("Registry");
+		let registry = await Registry.deploy();
+        await registry.deployed();
+
+        const Factory = await ethers.getContractFactory("RefunderFactory");
+		factory = await Factory.deploy(registry.address);
+		await factory.deployed();
+    });
+
+    it('Owner of Master Refunder should be deployer', async () => {
+        let masterRefunderOwner = await masterRefunder.owner();
+        expect(masterRefunderOwner).to.be.eq(owner.address, "Invalid master refunder owner");
+    });
```

Comment:
no need for initialising `master` refunder. We use the contract only for implementation logic